esphome/components/haier/climate.py

```diff
@@ -354,16 +409,22 @@ async def to_code(config):
     await uart.register_uart_device(var, config)
     await climate.register_climate(var, config)
 
-    if (CONF_WIFI_SIGNAL in config) and (config[CONF_WIFI_SIGNAL]):
+    if CONF_WIFI_SIGNAL in config:
         cg.add(var.set_send_wifi(config[CONF_WIFI_SIGNAL]))
```

Comment:
Since this defaults to False its always in the config now
```suggestion
    cg.add(var.set_send_wifi(config[CONF_WIFI_SIGNAL]))
```